src/core/namespace.c

```diff
@@ -201,6 +201,30 @@ static const MountEntry protect_system_strict_table[] = {
         { "/root",               READWRITE_IMPLICIT, true  },      /* ProtectHome= */
 };
 
+/*
+ * ProtectSystem=extra-strict table. In addition to protections of ProtectSystem=strict,
+ * execute permissions are removed anywhere except for /usr with MS_NOEXEC.
+ */
+static const MountEntry protect_system_extra_strict_table[] = {
+        { "/",                   READONLY,           false },
+        { "/proc",               READWRITE_IMPLICIT, false },      /* ProtectKernelTunables= */
+        { "/sys",                READWRITE_IMPLICIT, false },      /* ProtectKernelTunables= */
+        { "/dev",                READWRITE_IMPLICIT, false },      /* PrivateDevices= */
+        { "/home",               READWRITE_IMPLICIT, true  },      /* ProtectHome= */
+        { "/run/user",           READWRITE_IMPLICIT, true  },      /* ProtectHome= */
+        { "/root",               READWRITE_IMPLICIT, true  },      /* ProtectHome= */
+        { "/",                   NOEXEC,             false },
+        { "/usr",                EXEC,               false },
+#if HAVE_SPLIT_USR
+        { "/lib",                EXEC,               true  },
+        { "/lib64",              EXEC,               true  },
+        { "/bin",                EXEC,               true  },
+#  if HAVE_SPLIT_BIN
+        { "/sbin",               EXEC,               true  },
+#  endif
+#endif
```

Comment:
i probably wouldn't bother too much with this, an instead just disable the thing altogether on split-usr systems...

Reply:
(but if you want to keep this, i am also fine with that... i personally would just consider split usr systems legacy that we can't even test anymore, and which get no tweaks except the most basic baseline support. and the baseline support here would be to turn the etxra-strict stuff off, in those cases)